[ISV-4497] validate catalog format

mantomas opened the pull request:
JIRA: ISV-4497

Allda commented on operator-pipeline-images/operatorcert/entrypoints/validate_catalog_format.py:
```diff
@@ -0,0 +1,120 @@
+def run_makefile(makefile_path: str) -> None:
+    """Run makefile"""
+    cmd = [
+        "make",
```
The makefile target should be `make catalog` based on a template given to us by opm team.

mantomas replied:
Changed, thanks for pointing out.

mavaras reviewed (approved):
LGTM :+1:

mporrato commented on ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-hosted-pipeline.yml:
```diff
@@ -411,12 +411,36 @@ spec:
+          value: "$(tasks.detect-changes.results.affected_catalogs)"
+        - name: operator_path
+          value: "$(tasks.detect-changes.results.operator_path)"
+      workspaces:
+        - name: source
+          workspace: repository
+          subPath: src
```
We need to make sure the makefile cannot tamper with the files in the repository workspace.

mantomas commented:
Removed Makefile validation and changed task order to first validate the catalog format and then build the fragment image. Test pipeline can be seen [here](https://gist.github.com/rh-operator-bundle-bot/835d3ac2a5935a567f2e7436c65739b8).

mporrato reviewed (approved):
Since this is now a rather trivial test, I would have done it as a static check in the common suite, but this approach should work too.
